apps/ewallet/test/ewallet/web/paginator_test.exs

```diff
@@ -129,13 +180,121 @@ defmodule EWallet.Web.PaginatorTest do
                count: 3
              }
     end
+
+    test "returns correct pagination data when query by given %{`page_record_field`, `page_record_value`}" do
+      per_page = 10
+      total_records = 5
+
+      # Generate 10 accounts
+      # Example: [%{id: "acc_1"}, %{id: "acc_2"}, ... , %{id: "acc_10"}]
+      ensure_num_records(Account, 10)
+
+      # Fetch last `total_records` elements from db
+      # Example: [%{id: "acc_6"}, %{id: "acc_7"}, ... , %{id: "acc_10"}]
+      records_id = from(a in Account, select: a.id, order_by: a.id)
+
+      records_id =
+        records_id
+        |> Repo.all()
+        # get last 5 records
+        |> Enum.take(-total_records)
+
+      # Example "acc_6"
+      first_record_id = Enum.at(records_id, 0)
+
+      paginator =
+        Paginator.paginate(
+          Account,
+          %{
+            "page_record_field" => :id,
+            "page_record_value" => first_record_id,
+            "per_page" => per_page
+          }
+        )
+
+      # Collect id-mapped paginator.data
+      actual_records_id =
+        paginator.data
+        |> Enum.map(fn %Account{id: id} -> id end)
+
+      assert actual_records_id == records_id
+
+      assert paginator.pagination == %{
+               per_page: per_page,
+               current_page: 1,
+               is_first_page: true,
+               is_last_page: true,
+               count: total_records
+             }
+    end
+
+    test "returns correct pagination data when query by given %{`page`, `page_record_field`, `page_record_value`}" do
+      per_page = 5
+
+      # Generate 10 accounts
+      ensure_num_records(Account, 10)
```

Comment:
We usually use `insert_list` to insert multiple records

Reply:
Sorry I'm probably the cause of this confusion...

You can use `insert_list/3` to insert n number of new records. 

On the other hand, `ensure_num_records/2` checks how many records are already available, and inserts more if there's not enough. This one is useful for `/*.all` calls where it might include records from the `setup` too.

Reply:
I copied it from the other tests in the same file 😄